plugins/gatsby-remark-variables/index.js

```diff
@@ -0,0 +1,68 @@
+const visit = require("unist-util-visit");
+
+function scopedEval(expr, context = {}) {
+  const evaluator = Function.apply(null, [
+    ...Object.keys(context),
+    "expr",
+    "return eval(expr)",
+  ]);
+  return evaluator.apply(null, [...Object.values(context), expr]);
+}
+
+const matchEach = (text, pattern, callback) => {
+  let match, rv;
+  let promises = [];
+  while ((match = pattern.exec(text)) !== null) {
+    rv = callback(match);
+    if (rv instanceof Promise) {
+      promises.push(rv);
+    }
+  }
+  return Promise.all(promises);
+};
+
+module.exports = async ({ markdownAST, markdownNode }, options) => {
+  const page = markdownNode.frontmatter;
+
+  visit(
+    markdownAST,
+    () => true,
+    async node => {
+      // XXX(dcramer): by far the worst template language of all time
+
+      if (!node.value) {
+        return;
+      }
+
+      // TODO(dcramer): this could be improved by parsing the string piece by piece so you can
```

Comment:
this is the main thing to note here.. likely won't be a big deal in practice, but would be ideal to support. problem for another day unless someones bored.